staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -40,14 +48,26 @@ type CustomResourceDefinitionStorageStrategy struct {
 	validator       customResourceValidator
 }
 
-func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind) CustomResourceDefinitionStorageStrategy {
+func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind, crd *apiextensions.CustomResourceDefinition) CustomResourceDefinitionStorageStrategy {
+	schema := &spec.Schema{}
+	if err := apiservervalidation.ConvertToOpenAPITypes(crd, schema); err != nil {
+		utilruntime.HandleError(err)
```

Comment:
why not pass the openapi scheme here as parameter and do the conversion in the caller (getServingInfoFor?). There we can handle the error better than here.